src/services/date_notes.ts

```diff
@@ -33,7 +41,87 @@ const MONTH_TRANSLATION_IDS = [
     "months.december"
 ];
 
-type StartOfWeek = "monday" | "sunday";
+/** produces 1st, 2nd, 3rd, 4th, 21st, 31st for 1, 2, 3, 4, 21, 31 */
+function ordinal(dayNumber: number) {
+    const suffixes = ["th", "st", "nd", "rd"];
+    const suffix = suffixes[(dayNumber - 20) % 10] || suffixes[dayNumber] || suffixes[0];
+    return `${dayNumber}${suffix}`;
+}
```

Comment:
This is English-specific so we cannot use it. See if `dayjs` has something, since we can enable localisation for it (and we do, at least on the client side for the calendar).